memgpt/server/server.py

```diff
@@ -900,7 +905,7 @@ def list_agents_legacy(
         for agent_state, return_dict in zip(agents_states, agents_states_dicts):
 
             # Get the agent object (loaded in memory)
-            memgpt_agent = self._get_or_load_agent(user_id=user_id, agent_id=agent_state.id)
+            memgpt_agent = self._get_or_load_agent(user_id=agent_state.user_id, agent_id=agent_state.id)
```

Comment:
I think this makes more sense ; to use the current agent state's user_id vs global user_id